ndk/src/signers/private-key/index.ts

```diff
@@ -1,25 +1,45 @@
 import type { UnsignedEvent } from "nostr-tools";
-import { generatePrivateKey, getPublicKey, getSignature, nip04 } from "nostr-tools";
+import { generateSecretKey, getPublicKey, finalizeEvent, nip04 } from "nostr-tools";
 
 import type { NostrEvent } from "../../events/index.js";
 import { NDKUser } from "../../user";
 import type { NDKSigner } from "../index.js";
+import { hexToBytes } from "@noble/hashes/utils";
+import { nip19 } from "nostr-tools";
 
 export class NDKPrivateKeySigner implements NDKSigner {
     private _user: NDKUser | undefined;
-    privateKey?: string;
+    privateKey?: Uint8Array;
 
-    public constructor(privateKey?: string) {
+    public constructor(privateKey?: Uint8Array | string) {
```

Comment:
Handle backwards compatibility for creating new private key signers with hex encoded private keys (also handles creating a signer with an `nsec` encoded private key).